src/main/java/dude/Storage.java

```diff
@@ -0,0 +1,147 @@
+package dude;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Scanner;
+
+import dude.task.Deadline;
+import dude.task.Event;
+import dude.task.Task;
+import dude.task.Todo;
+
+/**
+ * Storage class that controls task storing functionalities of Dude.
+ */
+public class Storage {
+    private static final String directoryPath = "./data";
+    private String filePath = "./data/dude.txt";
+
+    /**
+     * Class constructor.
+     */
+    public Storage() {
+        boolean success = createStorageIfNotExists();
+        if (!success) {
+            throw new DudeNoStorageException();
+        }
+    }
+
+    /**
+     * Class constructor specifying the list name.
+     * @param listName List name which will be used as the path for the storage file.
+     */
+    public Storage(String listName) {
+        this.filePath = String.format("./data/%s.txt", listName);
+        boolean success = createStorageIfNotExists();
+        if (!success) {
+            throw new DudeNoStorageException();
+        }
+    }
+
+    private boolean createStorageIfNotExists() {
+        new File(Storage.directoryPath).mkdirs();
+        File storageFile = new File(this.filePath);
+        if (!storageFile.exists()) {
+            System.out.println("dude.Storage file not found. Creating storage file...");
+            try {
+                storageFile.createNewFile();
+            } catch (IOException e) {
+                System.out.println("An error occurred creating the storage file.");
+                System.out.println(e);
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Creates row in storage file.
+     * @param task The task to create a row of.
+     * @return Whether creation was successful.
+     */
+    public boolean createRow(Task task) {
+        try {
+            FileWriter fw = new FileWriter(this.filePath, true);
+            fw.write(task.getStorageString() + System.lineSeparator());
+            fw.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred writing to the storage file.");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Creates multiple rows in storage file (clears out whatever data was previously there).
+     * @param tasks The ArrayList of tasks to create rows of.
+     * @return Whether creation was successful.
+     */
+    public boolean createRows(ArrayList<Task> tasks) {
+        boolean success = this.clearStorage();
+        if (!success) {
```

Comment:
I love how you are throwing success / failure upwards for functions like this, however, upon failure, would it be better to throw a error from the lower components and handle it instead of just solely passing booleans?